.github/workflows/host-image-ci.yml

```diff
@@ -0,0 +1,355 @@
+# This file is a template, and might need editing before it works on your project.
+# This is a sample GitLab CI/CD configuration file that should run without any modifications.
+# It demonstrates a basic 3 stage CI/CD pipeline. Instead of real tests or scripts,
+# it uses echo commands to simulate the pipeline execution.
+#
+# A pipeline is composed of independent jobs that run scripts, grouped into stages.
+# Stages run in sequential order, but jobs within stages run in parallel.
+#
+# For more information, see: https://docs.gitlab.com/ee/ci/yaml/index.html#stages
+#
+# You can copy and paste this template into a new `.gitlab-ci.yml` file.
+# You should not add this template to an existing `.gitlab-ci.yml` file by using the `include:` keyword.
+#
+# To contribute improvements to CI/CD templates, please follow the Development guide at:
+# https://docs.gitlab.com/ee/development/cicd/templates.html
+# This specific template is located at:
+# https://gitlab.com/gitlab-org/gitlab/-/blob/master/lib/gitlab/ci/templates/Getting-Started.gitlab-ci.yml
+
+name: HostImage
+
+on: [pull_request, push]
+
+jobs:
+  build-installer-iso-job:
+    runs-on: ubuntu-latest
+    container:
+      image: debian@sha256:c2cedd7f80a4dd0f9f80d3699bd433ccf3de33ab63bfa2d4c4ba870c998222d6 # debian:bookworm-20240311
+    env:
+      DEBIAN_ISO_URL: https://deb.debian.org/debian/dists/bookworm/main/installer-arm64/current/images/netboot/mini.iso
+    steps:
+    - name: checkout repository
+      uses: actions/checkout@a81bbbf8298c0fa03ea29cdc473d45769f953675 # aka v2
+    - name: Prepare building environment
+      run: apt-get update && apt-get upgrade -y && apt-get install -y sudo && apt-get install -y wget libarchive-tools && apt-get install -y xorriso && apt-get install -y cpio xz-utils && apt-get install -y fdisk
+    - name: Inject name and ID into preseed 
+      run: echo "CI_PROJECT_NAME=${{ github.event.repository.name }}" >> cuttlefish-host-image-installer/preseed/after_install_1.sh && echo "CI_PIPELINE_ID=${{ github.repository}}/${{ github.workflow }}" >> cuttlefish-host-image-installer/preseed//after_install_1.sh
+    - name: Download Debian installer
+      run: cd cuttlefish-host-image-installer && wget -nv -c ${DEBIAN_ISO_URL} && ./addpreseed.sh && xz -9e preseed-mini.iso
+    - name: Publish preseed-mini.iso.xz
+      uses: actions/upload-artifact@v3
+      with:
+        name: preseed-mini
+        path: cuttlefish-host-image-installer/preseed-mini.iso.xz
+
+  build-cuttlefish-deb-job:
```

Comment:
This repeats the job we have on presubmit, can we reuse that?
https://docs.github.com/en/actions/using-workflows/reusing-workflows

Reply:
This debain package is for ARM64 and it uses cross-platform compile. But I agree that I should move this debain package build into presubmit, because presubmit is the one building cuttlefish debain packages.

Therefore, I moved ARM64 debain package build into presubmit, add workflow_call event (for reusable workflow) in presubmit, and use presubmit in my workflow.

Reply:
IIUC You don't need to run the entire presubmit workflow in this step, just the `build-arm64-cuttlefish-deb-job` job. Consider moving that job to a file on its own and use from both the presubmit workflow and this one.

Reply:
It's possible to depend on another workflow using `needs`, e.g. [`needs: [build-orchestrator]`](https://github.com/google/android-cuttlefish/blob/61c7bc37409dbbbf196599c8901d27483fe1f8df/.github/workflows/presubmit.yaml#L21).